go/vt/vttablet/onlineddl/executor.go

```diff
@@ -3586,6 +3586,29 @@ func (e *Executor) CompleteMigration(ctx context.Context, uuid string) (result *
 	return e.execQuery(ctx, query)
 }
 
+func (e *Executor) submittedMigrationConflictsWithPendingMigrationInSingletonContext(
+	ctx context.Context, submittedMigration, pendingOnlineDDL *schema.OnlineDDL,
+) (bool, error) {
+	if pendingOnlineDDL.MigrationContext == submittedMigration.MigrationContext {
+		// same migration context. this is obviously allowed
+		return false, nil
+	}
+	// Let's see if the pending migration is a revert:
+	revertedUUID, _ := pendingOnlineDDL.GetRevertUUID()
+	if revertedUUID == "" {
+		// Not a revert. So the pending migration definitely conflicts with our migration.
+		return true, nil
+	}
+
+	// The pending migration is a revert
+	if !pendingOnlineDDL.StrategySetting().IsSingletonContext() {
+		// Aha! So, our "conflict" is with a REVERT migration, which does _not_ have a -singleton-context
+		// flag. Because we want to allow REVERT migrations to run as concurrently as possible, we allow this scenario.
+		return false, nil
+	}
+	return true, nil
```

Comment:
What is the point of the error return? I don't see where that's used for anything as this is a binary/boolean call. Feels like it only complicates code on the caller side.

Reply:
Good catch! Must be a leftover from a previous flow. Removed `error` from return value